python/amici/ode_export.py

```diff
@@ -2643,6 +2862,22 @@ def _get_function_body(self,
                 lines.append(f'{function}[{index}] = '
                              f'{_print_with_exception(formula)};')
 
+        elif function in event_functions:
+            outer_cases = {}
+            for ie in range(self.model.num_events()):
+                inner_equations = equations[ie]
+                inner_lines = []
+                if not smart_is_zero_matrix(inner_equations):
+                    inner_cases = {
+                        ipar: _get_sym_lines_array(inner_equations[:, ipar],
+                                                   function, 0)
+                        for ipar in range(self.model.num_par())
+                        if not smart_is_zero_matrix(inner_equations[:, ipar])}
```

Comment:
`inner_equations` is checked for zero twice, first the complete one, then element-wise. not sure how costly. could do element-wise at first occurrence and save result.

Reply:
I was hoping that checking the whole thing is somewhat faster... But I removed the outer checking now.